### ISSUE TITLE
Descritive warning when 0 articles are found

### ISSUE THREAD

**Issue by voidplayer** [getpelican/pelican#2274]
```
CRITICAL: UndefinedError: list object has no element 0
```
Is the right error there. The template is trying to access what is not there

A helpful

```
NOTICE: 0 articles found
```

before processing the templates would be a nice abstract addition messages as the cause of the error is not very obvious unless you know quite a lot of the pelican internals

Probably the same about pages would be as useful

**Comment by avaris** [comment]
If you're trying to track down an issue I'd suggest running pelican with `--debug` option. It'll give you the exact traceback plus a whole lot more information.

Disabling plugins and using default theme can also be helpful.

Besides, I don't see an apparent link from info about "0 articles" to the error you get unless you know what's going on in the template at which point you'd have reasonable guess about the cause as well. And ultimately, this is a theme error and theme designers are expected to be somewhat familiar with the pelican internals.

**Comment by voidplayer** [comment]
Heres an example of what i meant:

To reproduce this, add WRITE_SELECTED = ('foo/bar')

```
$ make html
pelican /home/voidplayer/web/ -o /home/voidplayer//web/output -s /home/voidplayer/web/pelicanconf.py  -D
DEBUG: Pelican version: 3.7.1
DEBUG: Python version: 2.7.13
DEBUG: Adding current directory to system path
DEBUG: Temporarily adding PLUGIN_PATHS to system path
DEBUG: Loading plugin `subcategory`
DEBUG: Registering plugin `subcategory`
DEBUG: Loading plugin `gallery`
DEBUG: Registering plugin `gallery`
DEBUG: Restoring system path
DEBUG: Template list: [u'!simple/archives.html', u'!simple/article.html', u'!simple/author.html', u'!simple/authors.html', u'!simple/base.html', u'!simple/categories.html', u'!simple/category.html', u'!simple/gosquared.html', u'!simple/index.html', u'!simple/page.html', u'!simple/pagination.html', u'!simple/period_archives.html', u'!simple/tag.html', u'!simple/tags.html', u'!simple/translations.html', u'archives.html', u'article.html', u'author.html', u'authors.html', u'base.html', u'categories.html', u'category.html', u'gosquared.html', u'include/article_info.html', u'include/article_list.html', u'include/breadcrumb.html', u'include/cc-license.html', u'include/comments.html', u'include/comments_scripts.html', u'include/disqus_script.html', u'include/footer.html', u'include/ga.html', u'include/open_graph.html', u'include/toc.html', u'index.html', u'page.html', u'pagination.html', u'period_archives.html', u'subcategory.html', u'tag.html', u'tags.html', u'translations.html']
CRITICAL: UndefinedError: list object has no element 0
Traceback (most recent call last):
  File "/usr/bin/pelican", line 11, in <module>
    load_entry_point('pelican==3.7.1', 'console_scripts', 'pelican')()
  File "/usr/lib/python2.7/dist-packages/pelican/__init__.py", line 487, in main
    pelican.run()
  File "/usr/lib/python2.7/dist-packages/pelican/__init__.py", line 179, in run
    p.generate_output(writer)
  File "/usr/lib/python2.7/dist-packages/pelican/generators.py", line 600, in generate_output
    self.generate_pages(writer)
  File "/usr/lib/python2.7/dist-packages/pelican/generators.py", line 496, in generate_pages
    self.generate_direct_templates(write)
  File "/usr/lib/python2.7/dist-packages/pelican/generators.py", line 444, in generate_direct_templates
    page_name=os.path.splitext(save_as)[0])
  File "/usr/lib/python2.7/dist-packages/pelican/writers.py", line 224, in write_file
    page.save_as, override_output)
  File "/usr/lib/python2.7/dist-packages/pelican/writers.py", line 171, in _write_file
    output = template.render(localcontext)
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/kusanagi/stuff/projects/lucasweb/web/themes/pelican-twitchy/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/voidplayer/web/themes/pelican-twitchy/templates/base.html", line 322, in top-level template code
    {% include "include/footer.html" %}
  File "/home/voidplayer/web/themes/pelican-twitchy/templates/include/footer.html", line 1, in top-level template code
    {% set copy_date = (all_articles[0].date.strftime('%Y') if all_articles else articles[0].date.strftime('%Y')) %}
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 408, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: list object has no element 0
Makefile:65: recipe for target 'html' failed
make: *** [html] Error 1
```

As you can see, no reference to any failure about finding articles or pages

**Comment by voidplayer** [comment]
I can see now that this is probably a bug with the theme, which shouldnt access [0] without checking for it first

Still I just switched the theme, not a very advanced user or knew what I was looking at :/

**Comment by justinmayer** [comment]
Glad you found a solution, @voidplayer. Not sure there's much Pelican can do in certain cases where themes aren't implemented correctly.

**Comment by voidplayer** [comment]
I agree that the theme could check if articles exist, but a theme assuming theres articles is not too far fetched either for a blog generator like pelican :)

Anyway, the real problem was PEBCAK. I used WRITE_SELECTED = ('foo/bar.md') pointing to a missing article because of a typo, leading to having 0 articles

I had to dig for a long time to figure out what was causing the problem. I could had used some help from pelican to notice my error, like a notice log line for articles in WRITE_SELECTED not found like

```
NOTICE: WRITE_SELECTED 'foo/bar.md' not found
```
or 
```
NOTICE: 0 articles found
```
before trying pass them to the template and having the template crash and having to understand the backtrace :)

Thanks for pelican btw... im having a blast using it!
